Set method to manage error behavior and icon state

flo7842 opened the pull request:
Remove unuse method

Set method to fetch language code

Config error message

Fetch label value by key

Remove method AutoDisabledMode

Refactoring

Edit README

flo7842 committed:
Set method to manage error behavior and icon state


Remove unuse method

Set method to fetch language code


Config error message


Fetch label value by key


Remove method AutoDisabledMode


Refactoring


Edit README

flo7842 committed:
Merge branch 'master' into feature/popover_errors

flo7842 committed:
Merge branch 'master' into feature/popover_errors

flo7842 committed:
Correction after merge.

The labelsErrorFromLanguage.get(languageCode) method is from the java client update

flo7842 commented:
The labelsErrorFromLanguage.get(languageCode) method is from the java client update

flo7842 committed:
add method for error icon path

flo7842 committed:
Set icon path behavior

flo7842 committed:
Change method to fetch the content of i18n

flo7842 committed:
Change method to fetch I18n content

flo7842 committed:
Fetch label

flo7842 committed:
Change method to read label

flo7842 committed:
Fetching labels translation and errors translation

flo7842 committed:
Merge branch 'master' into feature/popover_errors

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualContentComponentBuilderImpl.java:
```diff
@@ -29,16 +34,22 @@
 
     @Inject
-    public ContextualContentComponentBuilderImpl(EdcClient edcClient, OpenUrlAction openUrlAction, Popover popover) {
+    public ContextualContentComponentBuilderImpl(EdcClient edcClient, OpenUrlAction openUrlAction, Popover popover, HttpReaderImpl httpReader) {
```
First: We inject the interface not the implementation
Second: This class is encapsulated in EdcClient and should never be used (and read the file for each popover is overkill)
Third: httpReader is never used, so removed it.

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualContentComponentBuilderImpl.java:
```diff
@@ -167,8 +191,40 @@ private JComponent getBody() throws IOException, InvalidUrlException {
+        return str.substring(0, ++position) + character + str.substring(position);
+    }
+
+    private JComponent getFailure() throws InvalidUrlException, IOException {
```
Return a default JComponent in case of exception. For example, a label which contains the text: "Error". This will avoid to get an empty label and nothing display in the popover.

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualContentComponentBuilderImpl.java:
```diff
@@ -212,4 +268,9 @@ private String getLabel(String key, String languageCode, String publicationId) t
+
+    private String getError(String key, String languageCode, String publicationId) throws IOException, InvalidUrlException {
+        LOGGER.debug("Getting error translation for key {}, language code: {}, publication id {}", key, languageCode, publicationId);
+        return this.edcClient.getError(key, languageCode, null);
```
why null?
Use the publicationId according to null.

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualTitleComponentBuilderImpl.java:
```diff
@@ -56,31 +76,73 @@
+    }
+
+    @Override
+    public JComponent build() throws InvalidUrlException, IOException {
```
In ContextualContentComponentBuilderImpl, you catch exception. Why do you expose them here?
Catch exception too...

greg7859 commented on src/main/java/fr/techad/edc/popover/builder/ContextualTitleComponentBuilder.java:
```diff
@@ -49,10 +53,26 @@
      * @return the contextual component
      */
-    T build();
+    T build() throws InvalidUrlException, IOException;
```
See remarks in impl

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/EdcSwingHelpImpl.java:
```diff
@@ -40,12 +42,24 @@ public JComponent createComponent(String mainKey, String subKey, String iconPath
-        JComponent component = contextualComponentBuilder.setKeys(mainKey, subKey, languageCode)
+        try {
+            enableContextItem = edcClient.getContextItem(mainKey, subKey, languageCode) != null;
+        } catch (InvalidUrlException | IOException e) {
```
ok, so that is the enableContextItem in this case?
Perhaps, you have to force the value: false?

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualTitleComponentBuilderImpl.java:
```diff
@@ -17,12 +26,23 @@
     private Font headerFontAttributes;
 
+    @Inject
+    public ContextualTitleComponentBuilderImpl(EdcClient edcClient, HttpReaderImpl httpReader) {
```
Same remark like before: don't use impl and remove the httpReader.

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualTitleComponentBuilderImpl.java:
```diff
@@ -1,13 +1,22 @@
 
+import com.google.common.collect.Sets;
+import fr.techad.edc.client.EdcClient;
+import fr.techad.edc.client.internal.TranslationConstants;
```
We should never use internal because it's not "visible" as public class. If this is visible, this contants enum should move in client.model.

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualTitleComponentBuilderImpl.java:
```diff
@@ -56,31 +76,73 @@
+        JLabel jLabelError = new JLabel();
+
+        if (this.showTitle || errorBehavior == ErrorBehavior.ERROR_SHOWN) {
+
```
remove empty line

flo7842 committed:
Correction of review changes

Removing Excption on build, removing duplicate method, adding Log to method, changing of the constants call, handling popover disabled

flo7842 committed:
Merge branch 'master' into feature/popover_errors

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualTitleComponentBuilderImpl.java:
```diff
@@ -62,25 +98,59 @@ public JComponent build() {
+        try {
+            container.add(getBody(), BorderLayout.CENTER);
+        } catch (InvalidUrlException e) {
+            e.printStackTrace();
```
use logger to display error

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualTitleComponentBuilderImpl.java:
```diff
@@ -62,25 +98,59 @@ public JComponent build() {
+        } catch (InvalidUrlException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
```
use logger to display error and you can use the multicatch

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/components/IconButtonListener.java:
```diff
@@ -65,10 +69,23 @@ public void mousePressed(MouseEvent e) {
+        try {
+            contextItem = edcClient.getContextItem(mainKey, subKey, helpConfiguration.getLanguageCode());
+        } catch (IOException ex) {
+            ex.printStackTrace();
```
use logger

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/components/IconButtonListener.java:
```diff
@@ -65,10 +69,23 @@ public void mousePressed(MouseEvent e) {
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        } catch (InvalidUrlException ex) {
+            ex.printStackTrace();
```
use logger and multicatch

flo7842 committed:
Adding Logger
